src/components/blog-list-client.tsx

```
@@ -205,101 +192,82 @@ export function BlogListClient({ posts }: BlogListClientProps) {
+          onClearTags={handleClearTags}
+          sortOption={sortOption}
+          onSortChange={handleSortChange}
+          viewMode={viewMode}
+          onViewModeChange={handleViewModeChange}
+          resultCount={filteredPosts.length}
+        />
 
         <main
           id="main-content"
-          className="container mx-auto px-4 max-w-7xl py-8 isolate"
+          role="main"
```

Biome (2.1.2)

[error] The elements with this role can be changed to the following elements:
<main>

For examples and more information, see WAI-ARIA Roles

(lint/a11y/useSemanticElements)